deployment/src/main/java/com/datastax/oss/quarkus/deployment/CassandraClientBuildTimeConfig.java

```diff
@@ -48,4 +48,17 @@
    */
   @ConfigItem(name = "metrics.node-enabled", defaultValue = "")
   public Optional<List<String>> metricsNodeEnabled;
+
+  /**
+   * The name of the algorithm used to compress protocol frames.
+   *
+   * <ul>
+   *   <li>lz4: requires net.jpountz.lz4:lz4 in the classpath.
+   *   <li>snappy: requires org.xerial.snappy:snappy-java in the classpath. Works only when not in
+   *       Native mode.
+   *   <li>the string "none" to indicate no compression - this is a default value.
+   * </ul>
+   */
+  @ConfigItem(name = "protocol.compression", defaultValue = "none")
+  public String protocolCompression;
```

Comment:
Do we want this to _only_ be specified at build time?  If we also make this support a run-time override we could add this to the runtime props in CassandraClientConnectionConfig (which seems like a fairly natural home for it anyways) and just set the value on the runtime config we pass as an arg to CassandraClientProcessor.configureRuntimeProperties().  The main advantage I see is that this would allow us to avoid the per-operation logic (recorder.configureCompression, recorder.configureMetricsRegistry, recoder.configureRuntimeProps); it seems like the way we're doing it now could snowball in complexity as we add new things to support here.

In short, I'm wondering if we can't consolidate most of this in a single representation (which naturally would be the runtime config) and avoid a fair amount of the ad-hoc CassandraClientRecorder customization here.

Reply:
all settings that are in build time are here because this information is needed at compile time. Also, according to the quarkus team, we should move as many settings as possible to build time (instead of runtime). Regarding:
>The main advantage I see is that this would allow us to avoid the per-operation logic (recorder.configureCompression, recorder.configureMetricsRegistry, recoder.configureRuntimeProps);

If that will become an issue, we can refactor setting build time config into common code. Right now we have only two methods so I think it will be over-design at this point of the project.

Reply:
I don't think we _need_ the information at build-time.  We don't do anything with it until we use the build-time config info to populate the PDCLB... but that happens at run-time.  It seems like we're moving it to build-time because we want to follow the Quarkus suggestion to support as much as we can at build-time... and I agree with that.

I'm not arguing to move it to run-time only.   I'm suggesting we could specify it at build-time and support a run-time override.  This approach would allow us to support something even more powerful; a unified configuration that lives in a single place.  That's really what I'm after.

This isn't necessarily something we need to address in this PR; I know there's other work to cover other bits of the configuration.  But it is something I'd argue we need to consider at some point in the future.

Reply:
We need that info on build time for registering reflection classes, see:
https://github.com/datastax/cassandra-quarkus/pull/6/files#diff-4d87f287c23fd7e2ebc1d0421519bf60R116